lib/stimulus_reflex/html/document.rb

```diff
@@ -7,20 +7,22 @@ class Document
 
       delegate :element, to: :@document
 
-      def to_html
-        @document.root.to_html(save_with: DEFAULT_HTML_WITHOUT_FORMAT)
+      def document_element
+        @document&.root
       end
 
       def outer_html
-        @document.root.to_html(save_with: DEFAULT_HTML_WITHOUT_FORMAT)
+        document_element ? document_element.to_html(save_with: DEFAULT_HTML_WITHOUT_FORMAT) : ""
       end
+      alias_method :to_html, :outer_html
 
       def inner_html
-        @document.root.inner_html(save_with: DEFAULT_HTML_WITHOUT_FORMAT)
+        document_element ? document_element.inner_html(save_with: DEFAULT_HTML_WITHOUT_FORMAT) : ""
       end
 
       def initialize(html)
-        @document = Nokogiri::HTML5::Document.parse(html.to_s)
+        @initial_html = html.to_s
+        @document = parsing_class.parse(@initial_html)
```

Comment:
I think we can even bring this back now
```suggestion
        @document = Nokogiri::HTML5::Document.parse(html.to_s)
```